sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/StateHolder.java

```diff
@@ -94,11 +104,89 @@ static boolean getLoadStateFeatureFlag(String name) {
     static void setLoadState(String name, Boolean loaded) {
         LOAD_STATE.put(name, loaded);
     }
-    
+
     /**
      * @param name the loadState feature flag name to set
      */
     static void setLoadStateFeatureFlag(String name, Boolean loaded) {
         setLoadState(name + FEATURE_ENDPOINT, loaded);
     }
+
+    /**
+     * @return the nextForcedRefresh
+     */
+    public static Instant getNextForcedRefresh() {
+        return nextForcedRefresh;
+    }
+
+    /**
+     * Set after load or refresh is successful.
+     * @param nextForcedRefresh the nextForcedRefresh to set
+     */
+    public static void setNextForcedRefresh(Duration refreshPeriod) {
+        nextForcedRefresh = Instant.now().plusSeconds(refreshPeriod.getSeconds());
+    }
+
+    /**
+     * Sets a minimum value until the next refresh. If a refresh interval has passed or is smaller than the calculated
+     * backoff time, the refresh interval is set to the backoff time.
+     * @param refreshInterval period between refreshe checks.
+     * @param properties Provider properties for min and max backoff periods.
+     */
+    static void updateNextRefreshTime(Duration refreshInterval, AppConfigurationProviderProperties properties) {
+        if (refreshInterval != null) {
+            Instant newForcedRefresh = getNextRefreshCheck(nextForcedRefresh,
+                clientRefreshAttempts, refreshInterval.getSeconds(), properties);
+
+            if (newForcedRefresh.compareTo(nextForcedRefresh) != 0) {
+                clientRefreshAttempts += 1;
+            }
+            nextForcedRefresh = newForcedRefresh;
+        }
+
+        for (Entry<String, State> entry : STATE.entrySet()) {
+            State state = entry.getValue();
+            Instant newRefresh = getNextRefreshCheck(state.getNextRefreshCheck(),
+                state.getRefreshAttempt(), (long) state.getRefreshInterval(), properties);
+
+            if (newRefresh.compareTo(entry.getValue().getNextRefreshCheck()) != 0) {
+                state.incrementRefreshAttempt();
```

Comment:
Same idea as https://github.com/Azure/azure-sdk-for-java/pull/26606/files#r832700256
Why increment refresh attempts only when newRefresh != oldRefresh?